File: backend/app/schemas/risk_profile.py
from typing import Literal

from pydantic import BaseModel


RiskLabel = Literal["Conservative", "Moderate", "Aggressive"]


class RiskDimensionScore(BaseModel):
    dimension: str
    score: int
    rationale: str


class RiskProfileResult(BaseModel):
    profile: RiskLabel
    total_score: int
    dimension_scores: list[RiskDimensionScore]
    summary: str

File: frontend/src/types/riskProfile.ts
export type RiskLevel = "Conservative" | "Moderate" | "Aggressive";

export type RiskDimensionScore = {
  dimension: string;
  score: number;
  rationale: string;
};

export type RiskProfileResponse = {
  profile: RiskLevel;
  total_score: number;
  dimension_scores: RiskDimensionScore[];
  summary: string;
};